### PR TITLE
Tests for FLEET-64 and FLEET-65 testing helmRepoURLRegex feature

### PR DESCRIPTION
**Pull request by thehejik**
### Two cases automated
* FLEET-64 is testing helmRepoURLRegex matching the helm repo URL via this [fleet.yaml](https://github.com/fleetqa/fleet-qa-examples-public/blob/main/helm-urlregex-repo/fleet.yaml)
* FLEET-65 is testing helmRepoURLRegex matching the full helm chart URL without involving helm repo via this [fleet.yaml](https://github.com/fleetqa/fleet-qa-examples-public/blob/main/helm-urlregex-chart/fleet.yaml)

Both tests are using positive (the regexp matches) and negative (regexp doesn't match) testing. The tests are designed to be working on **local cluster only** and will be executed **only on Rancher:2.9**.

Both cases are using local (deployed on the cluster under test) helm registry and simple helm chart stored there. The needed local helm registry and its resources are deployed by using new `importYaml` function.

Testrun on v2.9 scheduled here: https://github.com/rancher/fleet-e2e/actions/runs/9320045815
image

(The CI check triggered by this PR is irrelevant as the tests are not executed on v2.8)

### CODE REVIEW

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts [hunk 1]
```diff
@@ -270,6 +270,73 @@ if (!/\/2\.7/.test(Cypress.env('rancher_version'))) {
...
+            cy.fleetNamespaceToggle('fleet-local');
+            cy.addFleetGitRepo({ repoName, repoUrl, branch, path, gitOrHelmAuth, gitAuthType, userOrPublicKey, pwdOrPrivateKey, helmUrlRegex });
+            cy.clickButton('Create');
+            cy.verifyTableRow(0, 'Active', /([1-9]\d*)\/\1/);
+            cy.accesMenuSelection('local', 'Storage', 'ConfigMaps');
+            cy.nameSpaceMenuToggle('All Namespaces');
+            cy.filterInSearchBox('local-chart-configmap');
+            cy.wait(2000);
+            cy.get('.col-link-detail').contains('local-chart-configmap').should('be.visible').click({ force: true });
+            cy.get('section#data').should('contain', 'sample-cm').and('contain', 'sample-data-inside');
```
[the remark this answers is not in the document]
Comment (mmartin24) [reply tests/cypress/e2e/unit_tests/p1_fleet.spec.ts:326]: or is it this part?

> Private helm repository should be accessible from by validating helm private secrets.

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts [hunk 2]
```diff
@@ -270,6 +270,73 @@ if (!/\/2\.7/.test(Cypress.env('rancher_version'))) {
...
+        path: 'helm-urlregex-chart',
+        test_explanation: 'chart',
+        helmUrlRegex_matching: '^http.*app.*tgz$',
+      },
+    ]
+
+    // Actually just a preparation step
+    it("Prepare the private helm registry", { tags: '@preparation' }, () => {
+      cy.importYaml({ clusterName: 'local', yamlFilePath: 'assets/helm-server-with-auth-and-data.yaml' });
+      cy.nameSpaceMenuToggle('default');
```
Comment (mmartin24) [inline tests/cypress/e2e/unit_tests/p1_fleet.spec.ts:305]: Mind please reminding me why toggling to default is needed here instead to local?

**Review by mmartin24 — COMMENTED** [review]
PR looks ok, but pls review some questions and take a look at hardcoded qase number. Thanks

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (thehejik) [reply tests/cypress/e2e/unit_tests/p1_fleet.spec.ts:326]: thanks, I've fixed in qase

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts (hunk 2 above)
Comment (mmartin24) [reply tests/cypress/e2e/unit_tests/p1_fleet.spec.ts:305]: ok. just saw not so important. Feel free to leave it as this if you wish.

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts [hunk 3]
```diff
@@ -270,6 +270,73 @@ if (!/\/2\.7/.test(Cypress.env('rancher_version'))) {
...
+  describe('Private Helm Repository tests (helmRepoURLRegex)', { tags: '@p1'}, () => {
+
+    // So far using custom repo, we should expose fleet.yaml somewhere on fleet-qa-examples repo
+    const repoUrl = 'https://github.com/fleetqa/fleet-qa-examples-public.git'
+    const branch = 'main'
+    const userOrPublicKey = 'user'
+    const pwdOrPrivateKey = 'password'
+    const gitOrHelmAuth = 'Helm'
+    const gitAuthType = "http"
+    var helmUrlRegex
```
[the remark this answers is not in the document]
Comment (mmartin24) [reply tests/cypress/e2e/unit_tests/p1_fleet.spec.ts:285]: You are actually right with `const`, but `let` will work and I understand it is more secure as it is only available where it is defined. Anyhow, as you wish here.

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts (hunk 2 above)
Comment (thehejik) [reply tests/cypress/e2e/unit_tests/p1_fleet.spec.ts:304]: Lets keep it for now.

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (thehejik) [reply tests/cypress/e2e/unit_tests/p1_fleet.spec.ts]: Pushed a commit with `let`

**Review by mmartin24 — APPROVED** [review]
LGTM

File: tests/cypress/e2e/unit_tests/p1_fleet.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (thehejik) [reply tests/cypress/e2e/unit_tests/p1_fleet.spec.ts]: it works with `let`

**Pull request merged**